### PR TITLE
OP-434 use parameterized tests for MATERNITYRESTARTINJUNE

### PR DESCRIPTION
**Pull request by dbmalkovsky**
Use parameterized testing method for MATERNITYRESTARTINJUNE with `true` and `false` values.

### CODE REVIEW

File: src/test/java/org/isf/admission/test/Tests.java [hunk 1]
```diff
@@ -147,6 +166,14 @@ public void setUp() {
 		cleanH2InMemoryDb();
 	}
 
+	@Parameterized.Parameters(name ="Test with MATERNITYRESTARTINJUNE={0}")
+	public static Collection<Object[]> automaticlot() {
```
Comment (mwithi) [inline src/test/java/org/isf/admission/test/Tests.java:170]: wrong variable name

**Review by mwithi — CHANGES_REQUESTED** [review]
One variable has a wrong name.

Anyway I expect that some tests will fail because it is not always true that getNextYProg() returns YProg + 1.

Based on the MATERNINTYSTARTINJUNE flag it will if the admission date is before june, otherwise it will restart by 1 if it is the first admission in june.

I don't think we were testing in this way before but we should.

Thanks!

**Comment by dbmalkovsky** [comment]
I believe the current code will work for both values of MATERNINTYSTARTINJUNE now through the end of May of this year and then it will fail.

In AdmissionIoOperations.java the code is comparing **now** to **June**:
````
if (wardId.equalsIgnoreCase("M") && GeneralData.MATERNITYRESTARTINJUNE) 
   {
	if (now.get(Calendar.MONTH) < 6) 
````

I'll work on resetting the calendar in the test 😄

**Comment by dbmalkovsky** [comment]
I tried multiple ways to manipulate the dates for MATERNITYRESTARTINJUNE testing but the only way that seemed to work is by adding a new method and `public static` variables specific for testing in `AdmissionIoOperations`.     In the process I found that the original tests for `getNextYProg()` never found any records (because of date issues) and always just returned `1`.

It might not be elegant but it works.

File: src/main/java/org/isf/admission/service/AdmissionIoOperations.java [hunk 2]
```diff
@@ -273,6 +273,21 @@ public int getNextYProg(
...
+			return now;
+		}
+		// testing date June or later
+		if (afterJune) {
+			return new GregorianCalendar(now.get(Calendar.YEAR), 8, 1);
+		}
+		// testing data before June
+		return new GregorianCalendar(now.get(Calendar.YEAR), 0, 1);
+	}
+
```
Comment (mwithi) [inline src/main/java/org/isf/admission/service/AdmissionIoOperations.java:290]: we could add Javadoc to explain the testing purpose

**Review by mwithi — APPROVED** [review]
maybe some Javadoc could help.
it's ok otherwise.
```
Results :

Tests run: 1164, Failures: 0, Errors: 0, Skipped: 2
```

**Pull request merged**